cltk/phonology/old_english/orthophonology.py

```diff
@@ -0,0 +1,211 @@
+sh = Consonant(Place.post_alveolar, Manner.fricative, Voiced.neg, 'ʃ')
+ch = Consonant(Place.palatal, Manner.fricative, Voiced.neg, 'ç')
+x  = Consonant(Place.velar, Manner.fricative, Voiced.neg, 'x')
+y  = Consonant(Place.velar, Manner.fricative, Voiced.pos, 'ɣ')
+h  = Consonant(Place.glottal, Manner.fricative, Voiced.neg, 'h')
+
+# approximants
+l  = Consonant(Place.alveolar, Manner.approximant, Voiced.pos, 'l')
+l0 = Consonant(Place.alveolar, Manner.approximant, Voiced.neg, 'l̥')
+j  = Consonant(Place.palatal, Manner.approximant, Voiced.pos, 'j')
+w  = Consonant(Place.velar, Manner.approximant, Voiced.pos, 'w')
+w0 = Consonant(Place.velar, Manner.approximant, Voiced.neg, 'w')
+
+# rhotics
+r  = Consonant(Place.alveolar, Manner.approximant, Voiced.pos, 'r')
+r0 = Consonant(Place.alveolar, Manner.approximant, Voiced.neg, 'r̥')	
+
+
+### The Vowels
+
+# monophongs
+i  = Vowel(Height.close, Backness.front, Roundedness.neg, Length.short, 'i')
+i_long = i.lengthen()
+y  = Vowel(Height.close, Backness.front, Roundedness.pos, Length.short, 'y')
+y_long = y.lengthen()
+u  = Vowel(Height.close, Backness.back, Roundedness.pos, Length.short, 'u')
+u_long = u.lengthen()
+e  = Vowel(Height.mid, Backness.front, Roundedness.neg, Length.short, 'e')
+e_long = e.lengthen()
+oe = Vowel(Height.mid, Backness.front, Roundedness.pos, Length.short, 'ø')
+oe_long = oe.lengthen()
+o  = Vowel(Height.mid, Backness.back, Roundedness.pos, Length.short, 'o')
+o_long = o.lengthen()
+ae = Vowel(Height.open, Backness.front, Roundedness.neg, Length.short, 'æ')
+ae_long = ae.lengthen()
+a  = Vowel(Height.open, Backness.back, Roundedness.neg, Length.short, 'ɑ')
+a_long = a.lengthen()
+
+# diphthongs
+aea = ae + a
+ae_long_a = ae_long + a
+eo = e + o
+e_long_o = e_long + o
+iu = i + u
+i_long_u = i_long + u
+
+
+consonants = [
+	m, n, n0, ng,
+	p, b, t, d, k, g,
+	tsh, dsh,
+	f, v, th, dh, s, z, sh, ch, x, y, h,
+	l, l0, j, w, w0,
+	r, r0
+]
+
+vowels = [
+	i, i_long, y, y_long, u, u_long,
+	e, e_long, oe, oe_long, o, o_long,
+	ae, ae_long, a, a_long
+]
+
+diphthongs = [
+	aea, ae_long_a, eo, e_long_o, iu, i_long_u
+]
+
+sound_inventory = consonants + vowels + diphthongs
+
+# these are the UNCONDITIONED sounds of the letters.
+alphabet = {
+	'a' : a,
+	'ā' : a_long,
+	'æ' : ae,
+	'ǣ' : ae_long,
+	'b' : b,
+	'c' : k,
+	'ċ' : tsh,
+	'd' : d,
+	'ð' : dh,
+	'e' : e,
+	'ē' : e_long,
+	'f' : f,
+	'g' : g,
+	'ġ' : j,
+	'h' : h,
+	'i' : i,
+	'ī' : i_long,
+	'l' : l,
+	'm' : m,
+	'n' : n,
+	'o' : o,
+	'ō' : o_long,
+	'p' : p,
+	'r' : r,
+	's' : s,
+	't' : t,
+	'u' : u,
+	'ū' : u_long,
+	'w' : w,
+	'x' : x,
+	'y' : y,
+	'ȳ' : y_long,
+	'þ' : th
+}
+
+diphthongs_ipa = {
+	'ea' : aea,
+	'ēa' : ae_long_a,
+	'ie' : i,
+	'īe' : i_long,
+	'eo' : eo,
+	'ēo' : e_long_o,
+	'io' : iu,
+	'īo' : i_long_u
+}
+
+digraphs_ipa = {
+	'cg' : dsh,
+	'ng' : ng,
+	'sc' : sh
+}
+
+oe = Orthophonology(sound_inventory, alphabet, diphthongs_ipa, digraphs_ipa)

```

Comment:
It seems that these give us a reasonable coverage.  There are a couple of reasons why, from what I understand of the language:
(1) The orthography is quite transparent, because the alphabet, as adopted in the early middle ages, reflected current usage.  Modern English orthography instead, as we know, is a frightful mess of archaeology -- cool but hard to learn, and requiring loads of rules and exceptions.
(2) Because of geographic and historical variation in the phonology, researchers tend to stick, by convention, to Alfred's West Saxon dialect.  A deeper implementation would provide rules depending on the dialect, but I don't know that we know that much about other variants.